### ISSUE TITLE
VirtualBox appliance doesn't start graphic environment

### ISSUE THREAD

**Issue by paraseba** [NixOS/nixpkgs#16321]
## Issue description

Importing the Virtualbox appliance from https://nixos.org/releases/nixos/latest-ova-x86_64-linux and importing it into Virtualbox doesn't work as expected. It should launch an X11 environment but the window manager fails to start.

It's important because probably this is one of the main ways people give NixOS a try.
### Steps to reproduce

Follow the steps in https://nixos.org/nixos/download.html#virtualbox-appliances  start the VM. No graphic environment.

**Comment by aeosynth** [comment]
confirmed with `nixos-16.03.941.af412f2-x86_64-linux.ova`. there is no graphical login.

**Comment by alxempirical** [comment]
I also observed this.  Bizarrely. the third time I re-imported the `ova` into virtualbox, the KDE login manager appeared.

But `sudo nix-channel --update && sudo nixos-rebuild switch` kills the graphical enviroment!

The reason I was re-importing the `ova` was that after the above upgrade command, `sudo` stopped working.  I would get a message along the lines of `Who are you?  Unknown user demo.`  And I couldn't log back in.  This persisted across reboots.  But this failure stopped at the same time the KDE enviroment started working.

**Comment by 0w0** [comment]
Confirmed that, I install nixos as guest os on windows7 virtualbox5.0.22.

When I upgrade to `16.03.931.2f1174b`, X11 is failed.
but rollback to `16.03.902.a24903b`, it's fine again, some package might break.

nix-channel information：
```
~ $ sudo nix-channel --list
nixos https://nixos.org/channels/nixos-16.03
nixpkgs http://nixos.org/channels/nixpkgs-unstable
 ```

**Comment by paraseba** [comment]
@alxempirical there is a ticket for that demo user behavior  https://github.com/NixOS/nixpkgs/issues/14970

**Comment by domenkozar** [comment]
Fixed in 76f346d61d537f5bbe4f365f9f659df7024602b4 for 16.09, already fixed in master.
